File: src/main/java/com/server/ordering/domain/dto/FoodDto.java
package com.server.ordering.domain.dto;

import com.server.ordering.domain.Food;
import com.server.ordering.domain.RepresentativeMenu;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import static lombok.AccessLevel.PROTECTED;

@Getter @Setter
@AllArgsConstructor
@NoArgsConstructor(access = PROTECTED)
public class FoodDto {

    private String foodName;
    private int price;
    private String menuIntro;

    // response 전용
    private Long foodId;
    private String imageUrl;
    private boolean soldOut;

    public FoodDto(Food food) {
        this.foodId = food.getId();
        this.foodName = food.getFoodName();
        this.price = food.getPrice();
        this.soldOut = food.isSoldOut();
        this.imageUrl = food.getImageUrl();
        this.menuIntro = food.getMenuIntro();
    }

    public FoodDto(RepresentativeMenu menu) {
        this.foodId = menu.getFood().getId();
        this.foodName = menu.getFood().getFoodName();
        this.price = menu.getFood().getPrice();
        this.soldOut = menu.getFood().isSoldOut();
        this.imageUrl = menu.getFood().getImageUrl();
        this.menuIntro = menu.getFood().getMenuIntro();
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/dto/FoodDto.kt
package com.byeoru.ordering_server.domain.dto

import com.byeoru.ordering_server.domain.Food

class FoodDto {

    var foodName: String
    var price: Int
    var menuIntro: String

    // response 전용
    var foodId: Long?
    var imageUrl: String?
    var isSoldOut: Boolean

    constructor(foodName: String, price: Int, menuIntro: String, foodId: Long?, imageUrl: String?, soldOut: Boolean) {
        this.foodName = foodName
        this.price = price
        this.menuIntro = menuIntro
        this.foodId = foodId
        this.imageUrl = imageUrl
        this.isSoldOut = soldOut
    }

    constructor(food: Food) : this(
        food.foodName,
        food.price,
        food.menuIntro,
        food.id,
        food.imageUrl,
        food.isSoldOut
    )
}